Add some more documentation for devtools and OHOS debugging. (#83)
servo/book: src/hacking/debugging-on-openharmony.md

### Hilog domains

`hilog` allows setting a custom integer called "domain" (between 0 and 0xFFFF) when logging, which allows developers to easily filter logs using the domain.
For Rust code in servo logged via the `log` crate we set [`0xE0C3`] as the domain and for Spidermonkey C++ code we set [`0xE0C4`].
These values are somewhat arbitrarily chosen.

[`0xE0C3`]: https://github.com/servo/servo/blob/384d8f1ff895f070397b7a5a384428b1678416c5/ports/servoshell/egl/ohos.rs#L420
[`0xE0C4`]: <TODO - link pending SM PR being merged.>

### Hilog privacy feature

`hilog` has a privacy feature, which by default hides values in logs (e.g. from `%d` or `%s` substitutions).
Log statements from Rust are generally unaffected by this, since the string formatting is done on the Rust side.
If you encounter this issue when viewing C/C++ logs, you can temporarily turn off the privacy feature by running:

```shell
hdc shell hilog -p off
```

### Devtools and port forwards

> You can enable the devtools and connect with firefox remotely to them. It is easiest to do this with the command line via

Reviewer: ```suggestion

You can enable the devtools and connect to them remotely with Firefox. It is easiest to do this with the command line via
```

`hdc shell aa start -a EntryAbility -b org.servo.servo --psn=--devtools=1234`

> To connect to servo running you have to forward the port with

Reviewer: ```suggestion
To connect to an instance of Servo you have to forward the port with
```

`hdc fport tcp:1234 tcp:1234`. You should see a message that the forward succeeded. Now you can

> connect the devtools `localhost:1234`.

Reviewer: ```suggestion
connect to the devtools using `localhost:1234`.
```
